numpy/doc/shm.py

```diff
@@ -0,0 +1,187 @@
+"""
+
+====================
+Shared memory arrays
+====================
+
+Many numerical problems can be decomposed into sub-problems that can be solved
+in a parallel manner. Consider an example, when you have an array
+of random float values, and you want to convert them to a string
+and find out how many 9's are in them.
+
+There are various approaches:
+
+The multitasking-unaware approach
+=================================
+
+Normally, you would go for something like
+
+.. code-block:: python
+
+    import numpy as np
+
+    # We define the function
+    def howmany9(num):
+        string = str(num)
+        result = string.count("9")
+        return result
+
+    # We create another function from it, aware that the input parameter is an array
+    howmany9_vec = np.vectorize(howmany9)
+
+    # Our input data
+    in_arr = np.random.random(100)
+
+    # DEFINITIONS END HERE, now comes the computation.
+    # Output data follow
+    out_arr = howmany9_vec(in_arr)
+
+Although you can pass the array to a corresponding function, only one OS thread
+will work on it. That's not what we want, we want true multitasking
+(i.e. parallel execution).
+
+The threading approach
+======================
+
+Another option is to use threads.
+
+However, as long as you are using the CPython interpreter, it is going
+to stay the same even if you use the :mod:`threading` module due to a
+CPython feature called `GIL <https://wiki.python.org/moin/GlobalInterpreterLock>`_
+(global interpreter lock). GIL ensures that only one thread is active
+at a time, so threre is no true multitasking.
+
+.. code-block:: python
+
+    # Extra definitions to the multitasking-unaware setup -
+    # we have to prepare the output
+    out_arr = np.zeros(in_arr.shape, int)
+
+    # And finally the function wrapper.
+    # It can be a function defined deeper than in the module's top level.
+    def land_howmany9(index):
+        result = howmany9(in_arr[index])
+        out_arr[index] = result
+
+    # DEFINITIONS END HERE, now comes the computation.
+
+    import threading
+
+    for index in range(in_arr.size):
+        thread = threading.Thread(target=land_howmany9, args=(index,))
+        thread.start()
```

Comment:
I did that. Pity that the multiprocessing part of `futures` doesn't support the initializer that we know from `multiprocessing.Pool`, though.